__tests__/api.test.js

```diff
@@ -53,33 +53,91 @@ describe("GET:/api/articles", () => {
       .get("/api/articles/34890384")
       .expect(404)
       .then(({ body }) => {
-        expect(body.msg).toBe('bad request')
+        expect(body.msg).toBe("bad request");
       });
   });
   test("400: invalid id (string)", () => {
     return request(app)
       .get("/api/articles/sdsc")
       .expect(400)
       .then(({ body }) => {
-        expect(body.msg).toBe('bad request')
+        expect(body.msg).toBe("bad request");
+      });
+  });
+});
+
+describe("PATCH:/api/articles:article_id/", () => {
+  test("200:update votes on article by id (returns updated object)", () => {
+    return request(app)
+      .patch("/api/articles/6")
+      .expect(200)
+      .send({ inc_votes: 100 })
+      .then(({ body }) => {
+        expect(body.article).toEqual(
+          expect.objectContaining({
+            author: "icellusedkars",
+            title: "A",
+            topic: "mitch",
+            article_id: 6,
+            body: "Delicious tin of cat food",
+            created_at: "2020-10-18T01:00:00.000Z",
+            votes: 100,
+          })
+        );
+      });
+  });
+  test("404: invalid id doesnt exist(still a number)", () => {
+    return request(app)
+      .patch("/api/articles/34890384")
+      .expect(404)
+      .send({ inc_votes: 100 })
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found");
+      });
+  });
+  test("400: invalid id (string)", () => {
+    return request(app)
+      .patch("/api/articles/sdsc")
+      .send({ inc_votes: 100 })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+  test("400: body has object but no inc_votes key", () => {
+    return request(app)
+      .patch("/api/articles/sdsc")
+      .expect(400)
+      .send({banana : 100})
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+  test("400: body has inc_votes, but with other keys", () => {
+    return request(app)
+      .patch("/api/articles/sdsc")
+      .expect(400)
+      .send({inc_votes : 100,banana:'100'})
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
       });
```

Comment:
Think we're just missing a test for {inc_votes : 'invalid data type'}

i.e. not passed a number